client/src/app/formulaire/formulaire.component.spec.ts

```diff
@@ -33,20 +35,57 @@ describe('FormuleComponent', () => {
   }));
 
   it('formulaire est pas valide quand il est vide',waitForAsync(()=>{
-  expect(component.form.valid).toBeFalsy();
-  }));
+
+  expect(component.feedbackForm.valid).toBeFalsy();
+    
+    }));
   it('champs email est requis',waitForAsync(()=>{
-   let email = component.form.controls['email'];
+   let email = component.feedbackForm.controls['email'];
     expect(email.valid).toBeFalsy();
   }))
   it('champs email est valide quand il est remplis',waitForAsync(()=>{
-    let email = component.form.controls['email'];
+    let email = component.feedbackForm.controls['email'];
     email.setValue("binyat@gmail.com")
      expect(email.valid).toBeTruthy();
    }))
+   it('champs SendEmail est requis',waitForAsync(()=>{
+    let sendEmail = component.feedbackForm.controls['sendEmail'];
+     expect(sendEmail.valid).toBeFalsy();
+   }))
+   it('champs sendEmail est valide quand il est remplis',waitForAsync(()=>{
+     let sendEmail = component.feedbackForm.controls['sendEmail'];
+     sendEmail.setValue("bnyat.azizsharif@zenika.com")
+      expect(sendEmail!.valid).toBeTruthy();
+    }))
+    it('champs points positifs est requis',waitForAsync(()=>{
+      let pointsPositifs = component.feedbackForm.controls['pointsPositifs'];
+       expect(pointsPositifs.valid).toBeFalsy();
+     }))
+     it('champs points positifs est valide quand il est remplis',waitForAsync(()=>{
+       let pointsPositifs = component.feedbackForm.controls['pointsPositifs'];
+       pointsPositifs.setValue("les points positifs:.....")
+        expect(pointsPositifs.valid).toBeTruthy();
+      }))
+      it('champs axes ameliorations positifs est requis',waitForAsync(()=>{
+        let axesAmeliorations = component.feedbackForm.controls['axesAmeliorations'];
+         expect(axesAmeliorations.valid).toBeFalsy();
+       }))
+       it('champs axes ameliorations est valide quand il est remplis',waitForAsync(()=>{
+         let axesAmeliorations = component.feedbackForm.controls['axesAmeliorations'];
+         axesAmeliorations.setValue("les axes ameliorations:.....")
+          expect(axesAmeliorations.valid).toBeTruthy();
+        }))
```

Comment:
Pourquoi tu indente à chaque fois les it ?

Reply:
Parce-que comme t'as dit s'il y a une erreur on sait pas où elle se trompe. Sauf si on fait sorte genre un it pour tester l'invalidité du formulaire mais chaque champs différemment, pas comme je mettais cy.get(input:invalid).should('have.length',2). Au lieu je mets tous les cy.get('#email'),cy.get('#sendEmail)...,dans lt it() de vérifier l'invalidité. c'est ce que tu veux dire?

Reply:
La question, c'est pourquoi tu écrit : 
```
it("", () => {
   // test 1
   })
   it("", () => {
       // test 2
       })
       it("", () => {
         // test 3
         })
```

Il devrait être aligné.

Pour la question des multiples it, voir le commentaire précédent.

Reply:
yes c'est bien comme ça